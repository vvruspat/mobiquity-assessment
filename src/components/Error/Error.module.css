.error_text {
    color: var(--regular-color);
}