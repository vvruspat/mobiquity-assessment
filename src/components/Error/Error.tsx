import { FC, PropsWithChildren } from "react";

import { Card } from "../UIKit/Card";
import { Text } from "../UIKit/Text";

import s from "./Error.module.css";

type ErrorProps = {};

/**
 * Error component
 */
export const Error: FC<PropsWithChildren<ErrorProps>> = ({ children }) => {
  return (
    <Card isAccent={true} header={<Text size="m">Error</Text>}>
      <div className={s.error_text}>{children}</div>
    </Card>
  );
};
