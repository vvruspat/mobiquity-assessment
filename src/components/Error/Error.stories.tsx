import { ComponentStory, ComponentMeta } from "@storybook/react";

import { Error } from "./Error";
import "../../index.css";

export default {
  title: "Components/Error",
  component: Error,
} as ComponentMeta<typeof Error>;

const Template: ComponentStory<typeof Error> = (args) => (
  <Error {...args}>Unknown error</Error>
);

export const ErrorTemplate = Template.bind({});

ErrorTemplate.args = {};
