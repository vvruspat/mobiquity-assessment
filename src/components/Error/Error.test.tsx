import { render, screen } from "@testing-library/react";
import { Error } from "./Error";

test("renders error", () => {
  render(<Error>Message</Error>);

  const el = screen.getByTestId("card");

  expect(el).toHaveClass("card_accent");
  expect(el).toHaveTextContent("Message");
  expect(el).toHaveTextContent("Error");
  expect(el).toMatchSnapshot();
});
