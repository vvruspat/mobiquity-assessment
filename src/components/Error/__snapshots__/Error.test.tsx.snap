// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders error 1`] = `
<section
  class="card card_accent"
  data-testid="card"
>
  <header
    class="header"
  >
    <span
      class="text size_m regular"
      data-testid="text"
    >
      Error
    </span>
  </header>
  <div
    class="content"
  >
    <div
      class="error_text"
    >
      Message
    </div>
  </div>
</section>
`;
